components/compliance-service/compliance.go

```diff
@@ -444,22 +439,18 @@ func setup(ctx context.Context, connFactory *secureconn.Factory, conf config.Com
 
 	// set up the scanner, scheduler, and runner servers with needed clients
 	// these are all inspec-agent packages
-	scannerServer := scanner.New(mgrClient, nodesClient, db)
-	schedulerServer := scheduler.New(mgrClient, nodesClient, db, ingestClient, secretsClient, conf.RemoteInspecVersion)
-	runnerServer := runner.New(mgrClient, nodesClient, db, ingestClient, conf.RemoteInspecVersion)
+	scanner := scanner.New(mgrClient, nodesClient, db)
+	resolver := resolver.New(mgrClient, nodesClient, db, secretsClient)
+	runner.InitWorkflowManager(workflowManager, conf.InspecAgent.JobWorkers, ingestClient, scanner, resolver, conf.RemoteInspecVersion)
+
+	workflowManager.Start(ctx)
+	schedulerServer := scheduler.New(scanner, workflowManager)
 
 	// start polling for jobs with a recurrence schedule that are due to run.
 	// this function will sleep for one minute, then query the db for all jobs
 	// with recurrence and check if it's time to run the job
 	go schedulerServer.PollForJobs(ctx)
 
-	// start the InspecAgent workers.  the workers represent the amount of goroutines
-	// available to execute a scan.  the buffer size represents the maximum amount of jobs
-	// that may get backed up in the agent queue
-	logrus.Infof("compliance service initializing %d job workers with %d job buffer size",
-		conf.InspecAgent.JobWorkers, conf.InspecAgent.JobBufferSize)
-	runnerServer.SetWorkers(conf.InspecAgent.JobWorkers, conf.InspecAgent.JobBufferSize)
```

Comment:
fwiw, We think this was probably a no-op as it was setting  the worker count on the wrong runnerServer.